cmd/flakechecker/main.go

```diff
@@ -0,0 +1,157 @@
+	flag.BoolVar(&debug, "debug", false, "Enable debug log level")
+	versioninfo.AddFlag(flag.CommandLine)
+	flag.Parse()
+
+	if debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	err = p.run()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type recentFlakyTestInfo struct {
+	JobName   string
+	Name      string
+	Classname string
+	TotalAll  int
+	FailRatio int
+}
+
+func (p *flakeCheckerParams) checkFailedTests(bqClient biqQueryClient, failedTests []testcase.TestCase, flakeCheckerRecs []*flakeDetectionPolicy) error {
+	if len(failedTests) == 0 {
+		return errors.New(errDescNoFailedTests)
+	}
+
+	for _, failedTest := range failedTests {
+		found := false
+		log.Infof("Checking failed test: %q / %q / %q", p.jobName, failedTest.Name, failedTest.Classname)
+		for _, flakeCheckerRec := range flakeCheckerRecs {
+			match, err := flakeCheckerRec.matchJobName(p.jobName)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeCheckerRec.matchTestName(failedTest.Name)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeCheckerRec.matchClassname(failedTest.Classname)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			found = true
+			log.Infof("Match found: %q / %q / %q", flakeCheckerRec.config.MatchJobName, flakeCheckerRec.config.TestNameRegex, flakeCheckerRec.config.Classname)
+			totalRuns, failRatio, err := bqClient.GetRatioForTest(flakeCheckerRec, failedTest.Name)
+			if err != nil {
+				return errors.Wrap(err, errDescGetRatio)
+			}
+
+			if totalRuns < totalRunsLimit {
+				return errors.Wrap(fmt.Errorf("%d", totalRuns), errDescShortHistory)
+			}
+
+			if failRatio > flakeCheckerRec.config.RatioThreshold {
+				return errors.Wrap(fmt.Errorf("(%d > %d)", failRatio, flakeCheckerRec.config.RatioThreshold), errDescAboveThreshold)
+			}
+
+			log.Infof("Ratio is below threshold: (%d <= %d)", failRatio, flakeCheckerRec.config.RatioThreshold)
+		}
+
+		if !found {
+			return errors.Wrap(errors.New(failedTest.Name), errDescNoMatch)
+		}
+	}
+
+	return nil
+}
+
+func (p *flakeCheckerParams) run() error {
+	testSuites, err := junit.IngestDir(p.junitReportsDir)
+	if err != nil {
+		return errors.Wrap(err, "could not read files")
+	}
+
+	failedTests, err := testcase.GetFailedTests(testSuites)
+	if err != nil {
+		return errors.Wrap(err, "could not find failed tests")
+	}
+	log.Infof("Found %d failed tests", len(failedTests))
+
+	flakeConfigs, err := loadFlakeConfigFile(p.configFile)
+	if err != nil {
+		return errors.Wrapf(err, "unable to load config file (%s)", p.configFile)
+	}
+
+	bqClient, err := getNewBigQueryClient()
+	if err != nil {
+		return errors.Wrap(err, "unable to create BigQuery client")
+	}
+
+	if err = p.checkFailedTests(bqClient, failedTests, flakeConfigs); err != nil {
+		return errors.Wrap(err, "check failed tests")
+	}
+
+	log.Info("All failed tests are within allowed flake thresholds")
```

Comment:
```suggestion
	log.Info("All failed tests are within allowed flake thresholds.")
```

Reply:
We are logging everywhere without a period at the end. I'll leave it like that.